src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = [];
+
+  if (century) {
+    men = people.filter(person => person.sex === 'm'
+      && Math.ceil(person.died / 100) === century);
+  } else {
+    men = people.filter(person => person.sex === 'm');
+  }
+
+  const totalAge = men.reduce((total, person) =>
+    total + (person.died - person.born), 0);
+
+  return totalAge / men.length;
```

Comment:
You repeat this logic not once. So, create a separate function for calculating the average age